File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/ui/PlayerUiController.java
package com.pierfrancescosoffritti.androidyoutubeplayer.ui;

import android.graphics.drawable.Drawable;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.View;

import com.pierfrancescosoffritti.androidyoutubeplayer.ui.menu.YouTubePlayerMenu;


public interface PlayerUiController {
    PlayerUiController showUi(boolean show);
    PlayerUiController showPlayPauseButton(boolean show);

    PlayerUiController showVideoTitle(boolean show);
    PlayerUiController setVideoTitle(@NonNull String videoTitle);

    PlayerUiController enableLiveVideoUi(boolean enable);

    PlayerUiController setCustomAction1(@NonNull Drawable icon, @Nullable View.OnClickListener clickListener);
    PlayerUiController setCustomAction2(@NonNull Drawable icon, @Nullable View.OnClickListener clickListener);
    PlayerUiController showCustomAction1(boolean show);
    PlayerUiController showCustomAction2(boolean show);

    PlayerUiController showFullscreenButton(boolean show);
    PlayerUiController setFullScreenButtonClickListener(@NonNull View.OnClickListener customFullScreenButtonClickListener);

    PlayerUiController showMenuButton(boolean show);
    PlayerUiController setMenuButtonClickListener(@NonNull View.OnClickListener customMenuButtonClickListener);

    PlayerUiController showCurrentTime(boolean show);
    PlayerUiController showDuration(boolean show);

    PlayerUiController showSeekBar(boolean show);
    PlayerUiController showBufferingProgress(boolean show);

    PlayerUiController showYouTubeButton(boolean show);

    /**
     * Adds a View to the top of the player
     * @param view View to be added
     */
    PlayerUiController addView(@NonNull View view);

    /**
     * Removes a View added with {@link #addView addView}
     * @param view View to be removed
     */
    PlayerUiController removeView(@NonNull View view);

    @Nullable YouTubePlayerMenu getMenu();
    PlayerUiController setMenu(@NonNull YouTubePlayerMenu youTubePlayerMenu);
}

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/ui/PlayerUiController.kt
package com.pierfrancescosoffritti.androidyoutubeplayer.core.ui

import android.graphics.drawable.Drawable
import android.view.View

import com.pierfrancescosoffritti.androidyoutubeplayer.core.ui.menu.YouTubePlayerMenu


interface PlayerUiController {
    fun showUi(show: Boolean): PlayerUiController
    fun showPlayPauseButton(show: Boolean): PlayerUiController

    fun showVideoTitle(show: Boolean): PlayerUiController
    fun setVideoTitle(videoTitle: String): PlayerUiController

    fun enableLiveVideoUi(enable: Boolean): PlayerUiController

    /**
     * Set custom action to the left of the Play/Pause button
     */
    fun setCustomAction1(icon: Drawable, clickListener: View.OnClickListener?): PlayerUiController
    /**
     * Set custom action to the right of the Play/Pause button
     */
    fun setCustomAction2(icon: Drawable, clickListener: View.OnClickListener?): PlayerUiController
    fun showCustomAction1(show: Boolean): PlayerUiController
    fun showCustomAction2(show: Boolean): PlayerUiController

    fun showFullscreenButton(show: Boolean): PlayerUiController
    fun setFullScreenButtonClickListener(customFullScreenButtonClickListener: View.OnClickListener): PlayerUiController

    fun showMenuButton(show: Boolean): PlayerUiController
    fun setMenuButtonClickListener(customMenuButtonClickListener: View.OnClickListener): PlayerUiController

    fun showCurrentTime(show: Boolean): PlayerUiController
    fun showDuration(show: Boolean): PlayerUiController

    fun showSeekBar(show: Boolean): PlayerUiController
    fun showBufferingProgress(show: Boolean): PlayerUiController

    fun showYouTubeButton(show: Boolean): PlayerUiController

    /**
     * Adds a View to the top of the player
     * @param view View to be added
     */
    fun addView(view: View): PlayerUiController

    /**
     * Removes a View added with [PlayerUiController.addView]
     * @param view View to be removed
     */
    fun removeView(view: View): PlayerUiController

    fun getMenu(): YouTubePlayerMenu?
}
